onnxruntime/test/fuzzing/src/test.cpp

```diff
@@ -285,123 +280,113 @@ int main(int argc, char* argv[])
-            throw;
-          }
-          
-          // Update current time
+      // Initialize the model
+      //
+      if (model_proto.ParseFromIstream(&modelStream)) {
+        if (repoMode) {
+          Logger::testLog << L"Running Prediction for: " << modelFileName
+                          << L" with seed " << seed << Logger::endl;
+          mutateModelTest(model_proto, mutateModelDirName, userOpt, seed);
+          Logger::testLog << L"Finished Prediction for: " << modelFileName
+                          << L" with seed " << seed << Logger::endl;
+          return 0;
+        } else {
+          // Call the mutateModelTest
           //
-          currTime = std::chrono::system_clock::now();
-        }
-        Logger::testLog<< "Ending Test" << Logger::endl;
-        
-        if(userOpt.stress)
-        {
-          Logger::testLog.enable();
+          std::chrono::system_clock::time_point currTime{std::chrono::system_clock::now()};
+
+          std::chrono::minutes timeInMin{testTimeOut};
+          std::chrono::seconds timeInSec{testTimeOut};
+          std::chrono::hours timeInHrs{testTimeOut};
+          std::chrono::system_clock::time_point endTime{currTime};
+          endTime += scale == timeScale::Hrs ? timeInHrs
+                                             : scale == timeScale::Min ? timeInMin : timeInSec;
+          Logger::testLog << "Starting Test" << Logger::endl;
+          while (currTime < endTime) {
+            try {
+              onnx::ModelProto bad_model = model_proto;
+              Logger::testLog << "Starting Test iteration: " << iteration << Logger::endl;
+              mutateModelTest(bad_model, mutateModelDirName, userOpt);
+              num_successful_runs++;
+              Logger::testLog << "Completed Test iteration: " << iteration++ << Logger::endl;
+            } catch (...) {
+              fuzz_handle_exception();
+            }
+            // Update current time
+            //
+            currTime = std::chrono::system_clock::now();
+          }
         }
-
-        Logger::testLog<< L"Total number of exceptions: " << num_unknown_exception+num_std_exception+num_ort_exception << Logger::endl;
-        Logger::testLog<< L"Number of Unknown exceptions: " << num_unknown_exception << Logger::endl;
-        Logger::testLog<< L"Number of ort exceptions: " << num_ort_exception << Logger::endl;
-        Logger::testLog<< L"Number of std exceptions: " << num_std_exception << Logger::endl;
-        Logger::testLog << L"Number of unique errors: "<< werrStreamBuf.get_unique_errors() << L"\n";
-        
-        if(userOpt.stress)
-        {
-          Logger::testLog.disable();
-          Logger::testLog.flush();
+      } else {
+        throw std::exception("Unable to initialize the Onnx model in memory");
+      }
+    } else {
+      std::wstring ortModelFile = modelFile + L".ort";
+      Ort::SessionOptions so;
```

Comment:
Shouldn't we just use the specified model filename as is?

Reply:
The model_file is the input ONNX file. We are making sure that we run this test using the ORT file generated from the ONNX file so that we use the latest changes, instead of a save ORT checked in to the repo. The ort_model file name generated by adding ".ort" to the input ONNX filename.

---
In reply to: [521690308](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r521690308) [](ancestors = 521690308)

Reply:
This code does not generate the .ort file does it? If this code is just testing the input it is provided, why is it imposing some arbitrary rule about file naming that isn't visible to a user (i.e. the input should be an ort file that must have an .ort extension, but that extension should not be passed when running the program). It may make sense if we were always testing a pair of files, but afaik each test is a standalone test of a single input, be that an ONNX or an ORT format model.

Reply:
Because of the option kOrtSessionOptionsConfigSaveModelFormat and SetOptimizedModelPath the first session dump the ORT model to the specified file. The ortModelStream reads the dumped ORT file into model_data. We want to make sure sure that we are also using the latest changes/fixes in onnxruntime to save the or ORT model from ONNX rather than an outdate ORT file.

---
In reply to: [522423493](https://github.com/microsoft/onnxruntime/pull/5771#discussion_r522423493) [](ancestors = 522423493)